dev_guide/builds.adoc

```diff
@@ -1556,3 +1556,109 @@ Docker and Source builds set the following labels on output images:
 |*io.openshift.build.source-location*
 |Source URL for the build
 |===
+
+
+[[using-external-artifacts]]
+= Using External Artifacts During The Build
+
+In order to use external artifacts during the build you don't need any special
+support from OpenShift platform. You could download and use any artifacts
+as you want.
+
+For `*Source*` build strategy you need to put appropriate shell commands into
+the `assemble` script.
+
+.`.sti/bin/assemble` file
+====
+
+[source,bash]
+----
+#!/bin/sh
+APP_VERSION=1.0
+wget http://repository.example.com/app/app-$APP_VERSION.jar -O app.jar
+----
+====
+
+.`.sti/bin/run` file
+====
+
+[source,bash]
+----
+#!/bin/sh
+exec java -jar app.jar
+----
+====
+
+For `*Docker*` build strategy you need to modify `Dockerfile` and invoke shell
+commands with https://docs.docker.com/engine/reference/builder/#run[`RUN` instruction].
+
+.Excerpt of `Dockerfile`
+====
+
+[source]
+----
+FROM jboss/base-jdk:8
+
+ENV APP_VERSION 1.0
+RUN wget http://repository.example.com/app/app-$APP_VERSION.jar -O app.jar
+
+EXPOSE 8080
+CMD [ "java", "-jar", "app.jar" ]
+----
+====
+
+Both of the examples will work, but you could improve it by replacing local
+variable by the environment variable.
+
+Now let’s recall and summarize how to manage these environment variables.
+If you are using `*Source*` strategy you could choose between the following ways:
+
+.  link:#environment-files[put them into the `.sti/environment` file]
++
+[source,shell]
+----
+APP_VERSION=1.0
+----
+
+.  link:#buildconfig-environment[set them in `*BuildConfig*`]
++
+[source,yaml]
+----
+  strategy:
+    sourceStrategy:
+      env:
+      - name: APP_VERSION
+        value: "1.0"
+----
+
+.  provide them explicitly with `*oc start-build --env*` (only for builds that are triggered manually)
++
+[source,shell]
+----
+oc start-build --env=APP_VERSION=1.0 <BuildConfigName>
+----
+
+If you can't modify source repository and put `assemble` script there, consider
+one more way: publish custom version of the `assemble` script to some server and
+link:#override-builder-image-scripts[configure `BuildConfig` to download and use
+this script instead of the default or from the repository].
+
+For the `*Docker*` strategy there are only two options:
+
```

Comment:
the content after this line looks like a cut+paste of env discussion, not dockerfile overriding.

in anycase move the explanation of how to override the dockerfile directly below the example dockerfile, and if there is appropriate documentation that already exists, link to it rather than re-summarizing here.